File: src/main/scala/battleship/entity/Coordinate.scala
package battleship.entity

case class Coordinate(vertical: Int, horizontal: Char)

object VerticalPosition extends Enumeration {
  val one = Value("1")
  val two = Value("2")
  val three = Value("3")
  val four = Value("4")
  val five = Value("5")
  val six = Value("6")
  val seven = Value("7")
  val eight = Value("8")
  val nine = Value("9")
  val ten = Value("10")
}

object HorizontalPosition extends Enumeration {
  val a = Value("A")
  val b = Value("B")
  val c = Value("C")
  val d = Value("D")
  val e = Value("E")
  val f = Value("F")
  val g = Value("G")
  val h = Value("H")
  val i = Value("I")
  val j = Value("J")
}

File: src/main/java/battleship/entity/Coordinate.java
package battleship.entity;

import java.util.Objects;

public class Coordinate {

    private final int horizontal;
    private final int vertical;

    public Coordinate(int horizontal, int vertical) {
        this.horizontal = horizontal;
        this.vertical = vertical;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Coordinate that = (Coordinate) o;
        return horizontal == that.horizontal &&
                vertical == that.vertical;
    }

    @Override
    public int hashCode() {
        return Objects.hash(horizontal, vertical);
    }

    @Override
    public String toString() {
        return "Coordinate{" +
                "horizontal=" + horizontal +
                ", vertical=" + vertical +
                '}';
    }

    public int getHorizontal() {
        return horizontal;
    }

    public int getVertical() {
        return vertical;
    }
}
